src/query/tsdb/remote/server.go

```diff
@@ -153,18 +173,23 @@ func (s *grpcServer) Search(
 		return err
 	}
 
-	response, err := encodeToCompressedSearchResult(results, pools)
-	if err != nil {
-		logger.Error("unable to encode search result", zap.Error(err))
-		return err
-	}
+	size := min(defaultBatch, len(results))
+	for ; len(results) > size; results = results[size:] {
+		response, err := encodeToCompressedSearchResult(results, pools)
+		if err != nil {
+			logger.Error("unable to encode search result", zap.Error(err))
+			return err
+		}
 
-	err = stream.SendMsg(response)
-	if err != nil {
-		logger.Error("unable to send search result", zap.Error(err))
+		err = stream.Send(response)
+		if err != nil {
+			logger.Error("unable to send search result", zap.Error(err))
```

Comment:
Here too, should `return err` here if send fails, it's quite likely the connection is broken and should just terminate the send.